vshard/storage/init.lua

```diff
@@ -3883,7 +3883,13 @@ local function storage_cfg_xc(cfgctx)
         error('Error injection: cfg')
     end
 
-    if not cfgctx.is_reload then
+    if cfgctx.new_cfg.box_cfg_mode == 'manual' then
+        if type(box.cfg) == 'function' then
+            local msg = "Box must be configured, when box_cfg_mode is 'manual'"
+            error(lerror.vshard(lerror.code.STORAGE_IS_DISABLED, msg))
+        end
+        log.info("Box configuration was skipped")
```

Comment:
Lets add "due to the manual `box_cfg_mode`" just for verbosity. I also suggest to move this entire check as early as we can. Otherwise we create the new replicaset objects just to fail here without using them. Lets do this right after `log.info("Starting configuration/reconfiguration...`.

Reply:
Done